cpp/ql/lib/semmle/code/cpp/ir/implementation/raw/internal/TranslatedElement.qll

```diff
@@ -40,12 +40,28 @@ IRTempVariable getIRTempVariable(Locatable ast, TempVariableTag tag) {
   result.getTag() = tag
 }
 
+private Expr getAnOperand(BinaryOperation binOp) { result = binOp.getAnOperand() }
+
+private int getNumberOfBinaryOperands(BinaryOperation binOp) {
```

Comment:
I expect a binary operation to always have exactly two operands 😅, so the meaning is slightly different here?

Reply:
Oh, good point. I'll add some QLDoc here. The meaning is that `getNumberOfBinaryOperands` grabs all the operations recursively. So `getNumberOfBinaryOperands((1 + 2) + 3)` is `3` even though the outermost binary operation only has two operands (i.e., `1 + 2` and `3`).

Reply:
I've added some explanatory QLDoc in a97891cbc131bb7fc94fbede286832a675db2cba, and renamed the predicate to `getNumberOfNestedBinaryOperands` to make it clear that we're not just computing the number 2 😂